pilot/pkg/xds/endpoint_builder.go

```diff
@@ -286,6 +298,10 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep)
 			}
 			locLbEps.append(ep.EnvoyEndpoint, ep.TunnelAbility)
+
+			// detect if mTLS is possible for this endpoint, used later during ep filtering
+			// this must be done while converting IstioEndpoints because we still have workload labels
+			b.mtlsChecker.checkEndpoint(ep)
```

Comment:
Given this has performance impacts as high as 25% in our current benchmarks, and we only use the result if `b.MultiNetworkConfigured`, should we make this condition on `b.MultiNetworkConfigured`?